ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenRunner.java

```diff
@@ -183,24 +183,31 @@ public Void visitLikePredicate(final LikePredicate node, final TypeContext conte
 
     @Override
     public Void visitFunctionCall(final FunctionCall node, final TypeContext context) {
-      final List<SqlArgument> argumentTypes = new ArrayList<>();
       final FunctionName functionName = node.getName();
+
+      final TypeContext contextCopy = context.getCopy();
```

Comment:
nit: maybe rename to `lambdaSharedContext`?

Reply:
took me a while as well to understand what was going on. I'd name them like this:
```
// this is the one that's passed in, it represents the context at the time the parent called this method
TypeContext parentTypeContext; 

// this is a copy of the parent type context, additionally populated with the result of visiting non-lambda
// expressions - it starts as a copy of the parent typeContext
TypeContext currentTypeContext; 

// a copy of either parent or current type context to be passed to the child - in the case of lambdas
// we pass in the parent type context without the result of resolving the current context because 
// there may be valid overlapping lambda parameter names
TypeContext childContext;
```